scripts/restart-server

```diff
@@ -89,6 +89,17 @@ if has_application_server():
 if has_process_fts_updates():
     workers.append("process-fts-updates")
 
+# Before we start (re)starting main services, make sure that auxiliary
+# services that we don't stop, but expect to be running, actually are.
+aux_services = list_supervisor_processes("go-camo", "smokescreen")
+if aux_services:
+    # Hide stdout because otherwise `go-camo: ERROR (already started)`
+    # might be confusing, as it's not an error.
+    subprocess.check_call(
+        ["supervisorctl", "start", *aux_services],
+        stdout=subprocess.DEVNULL,
+    )
```

Comment:
I guess this may be the best we can readily do, but I don't like eating error output.  Is it worth opening an upstream issue with supervisord to offer a "start if not already running" flag, which we would at least be able to use eventually?

Reply:
`list_supervisor_processes` already parses the output of `supervisorctl status` to exclude nonexistent processes. We could add a way to exclude running processes too.

We should consider migrating to the [API](http://supervisord.org/api.html) so we don’t have to rely on `supervisorctl`’s output formatting.

Reply:
[Some minor hoops](https://stackoverflow.com/a/51377201) are required to use the on-disk socket.  There's some complexity that shelling out to supervisorctl does get for us (e.g. figuring out where the socket is, useful error handling, etc) but there's certainly something to be said for not doing more parsing of its output.

Reply:
Hmm, yeah, the socket approach doesn't seem super convenient here.  Do we want to extend `list_supervisor_processes` to support excluding running processes with a given parameter?

Reply:
It's not _so_ inconvenient.  Repushed using it.